pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,629 @@
+    // containers. For example the files might be newline seperated JSON
+    // structured logs, systemd-journald journal files, gRPC trace files, etc.
+    // E.g.,
+    //     PodSandboxConfig.LogDirectory = `/var/log/pods/<podUID>/`
+    //     ContainerConfig.LogPath = `containerName_Instance#.log`
+    //
+    // WARNING: Log managment and how kubelet should interface with the
+    // container logs are under active discussion in
+    // https://issues.k8s.io/24677. There *may* be future change of direction
+    // for logging as the discussion carries on.
+    required string log_directory = 3;
+    // DNSOptions sets the DNS options for the sandbox.
+    optional DNSOption dns_options = 4;
+    // PortMappings lists the port mappings for the sandbox.
+    repeated PortMapping port_mappings = 5;
+    // Resources specifies the resource limits for the sandbox (i.e., the
+    // aggregate cpu/memory resources limits of all containers).
+    // Note: On a Linux host, kubelet will create a pod-level cgroup and pass
+    // it as the cgroup parent for the PodSandbox. For some runtimes, this is
+    // sufficent. For others, e.g., hypervisor-based runtimes, explicit
+    // resource limits for the sandbox are needed at creation time.
+    optional PodSandboxResources resources  = 6;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 7;
+    // Annotations is an unstructured key value map that may be set by external
+    // tools to store and retrieve arbitrary metadata.
+    map<string, string> annotations = 8;
+    // Linux contains configurations specific to Linux hosts.
+    optional LinuxPodSandboxConfig linux = 9;
+}
+
+// DNSOption specifies the DNS servers and search domains.
+message DNSOption {
+    // Servers is a list of DNS servers of the cluster.
+    repeated string servers = 1;
+    // Searches is a list of DNS search domains of the cluster.
+    repeated string searches = 2;
+}
+
+// PortMapping specifies the port mapping configurations of sandbox
+message PortMapping {
+    // Name of the port mapping
+    optional string name = 1;
+    // Protocol of the port mapping.
+    // Valid candidates are: TCP, UDP
+    required string protocol = 2;
+    // The port number within the container.
+    required int32 container_port = 3;
+    // The port number on the host.
+    optional int32 host_port = 4;
+    // The host IP.
+    optional string host_ip = 5;
+}
+
+// Mount specifies the volume mount for the sandbox
+message Mount {
+    // Name of the volume mount.
+    required string name = 1;
+    // Path of the mount within the container.
+    required string container_path = 2;
+    // Path of the mount on the host.
+    required string host_path = 3;
+    // Whether the mount is read-only.
+    required bool readonly = 4;
+    // Whether the mount needs SELinux relabeling
+    optional bool selinux_relabel = 5;
```

Comment:
Hmm...this seems like an implementation detail that runtime can derive by looking at SELinuxOptions. WDYT?